source/Esf.Website/App/esfState/esfState.component.html

```diff
@@ -3,8 +3,25 @@
         <div class="navbar-header">
             <a class="navbar-brand" href="#">ESFiddle</a>
         </div>
-        <button type="button" (click)="queryRunner.run(state)" class="run-command btn btn-default navbar-btn">Run</button>
-        <button type="button" (click)="save()" class="save-command btn btn-default navbar-btn">Save</button>
+        <button type="button" 
+                (click)="runQueryCommand.run()" 
+                [ngClass]="{'disabled' : runQueryCommandState > 0}"
```

Comment:
I would use boolean checks for the state instead of a single enum. For example inProgress, success and only one of them will be true